basex-core/src/main/java/org/basex/core/locks/DBLocking.java

```diff
@@ -138,7 +138,9 @@ public void acquire(final Proc pr, final StringList read, final StringList write
       }
       // global read locking
       if(read == null) {
-        while(localWriters > 0) {
+        while(localWriters > 0 &&
+            // We're the only writer, allow global read lock anyway
+            !(1 == localWriters && !(null == write || write.isEmpty()))) {
```

Comment:
Yes, this seems reasonable to me. In other words:

1. There are _no_ `localWriters` at all (this was already allowed)
2. There is _exactly one_ `localWriter`, and the current query is a `localWriter`, so there cannot be any other.